src/utils/helper-functions/utils.ts

```diff
@@ -11,63 +10,100 @@ import {
 } from './guards'
 
 export const MODULE_PATHS = {
+  componentStore: '@ngrx/component-store',
   effects: '@ngrx/effects',
+  store: '@ngrx/store',
 }
 
-export function findClassDeclarationNode(
-  node: TSESTree.Node,
-): TSESTree.ClassDeclaration | null {
-  if (isClassDeclaration(node)) {
-    return node
+export function getNearestUpperNodeFrom<T extends TSESTree.Node>(
+  { parent }: TSESTree.Node,
+  predicate: (parent: TSESTree.Node) => parent is T,
+): T | undefined {
+  while (parent && !isProgram(parent)) {
+    if (predicate(parent)) {
+      return parent
+    }
+
+    parent = parent.parent
   }
-  if (!node.parent) return null
-  return findClassDeclarationNode(node.parent)
+
+  return undefined
 }
 
-export function findImportDeclarationNode(
+export function getImportDeclarationSpecifier(
+  importDeclarations: readonly TSESTree.ImportDeclaration[],
+  importedName: string,
+) {
+  for (const importDeclaration of importDeclarations) {
+    const importSpecifier = importDeclaration.specifiers.find(
+      (importClause): importClause is TSESTree.ImportSpecifier => {
+        return (
+          isImportSpecifier(importClause) &&
+          importClause.imported.name === importedName
+        )
+      },
+    )
+
+    if (importSpecifier) {
+      return { importDeclaration, importSpecifier } as const
+    }
+  }
+
+  return undefined
+}
+
+export function getImportDeclarations(
   node: TSESTree.Node,
-  module: string,
-): TSESTree.ImportDeclaration | undefined {
-  if (isProgram(node)) {
-    return (node.body || []).find(
-      (node) => isImportDeclaration(node) && node.source.value === module,
-    ) as TSESTree.ImportDeclaration | undefined
+  moduleName: string,
+): readonly TSESTree.ImportDeclaration[] | undefined {
+  let parentNode: TSESTree.Node | undefined = node
+
+  while (parentNode && !isProgram(parentNode)) {
+    parentNode = parentNode.parent
   }
-  if (!node.parent) return undefined
-  return findImportDeclarationNode(node.parent, module)
+
+  return parentNode?.body.filter(
+    (node): node is TSESTree.ImportDeclaration =>
+      isImportDeclaration(node) && node.source.value === moduleName,
+  )
 }
 
-export function getConditionalImportFix(
-  fixer: TSESLint.RuleFixer,
+export function getImportAddFix(
```

Comment:
Just noticed that this implementation is problematic for default imports, namespace imports and type-only imports, as it could potentially introduce a break code, i.e.:

```ts
import * as ngrx from '@ngrx/store'

// would become something like

import * as ngrx, Effect from '@ngrx/store'

```

...let me try something better (though it's broken in the current version also).

Reply:
It should be good now.